pinecone/data/index.py

```diff
@@ -180,7 +180,7 @@ def upsert(
         return UpsertResponse(upserted_count=total_upserted)
 
     def _upsert_batch(
-        self, vectors: List[Vector], namespace: Optional[str], _check_type: bool, **kwargs
+        self, vectors: Union[List[Vector], List[tuple], List[dict]], namespace: Optional[str], _check_type: bool, **kwargs
```

Comment:
What's the reason for this change here?

Reply:
The `upsert()` method that calls this `_upsert_batch()` takes a wide variety of user inputs (`Vector`, `tuple`, or `dict` representations of vectors) which aren't normalized to`Vector` until passed through the `VectorBuilder` referenced a few lines into this method. So `List[Vector]` is not a correct description of the input types accepted.

Reply:
Ok got it thanks!